torch/onnx/_internal/fx/fx_exporter.py

```diff
@@ -29,15 +29,23 @@ def export_fx_to_onnx(
             self.decomposition_table,
             enable_dynamic_axes=self.options.dynamic_shapes,
         ).run(*fx_module_args)
-
+        # Run FakeTensorProp to get fixed shape of nodes for op_level_debug purposes.
+        # NOTE: torch.fx.Transformer makes a copy of the graph only but shares weights
+        # with the original module.
+        static_reference_graph_module = torch.fx.Transformer(
+            decomposed_module
+        ).transform()
+        static_reference_graph_module = passes.ShapeInferenceWithFakeTensor(
```

Comment:
Ideally I would like to merge these operations into one pass. That pass runs these steps internally. And what's returned is still the `decomposed_module`, but with extra information on static tensor shapes. We'd need a definition for exporter metadata.

That way the implementation details are hidden, and you won't need to pass two graph modules to `export_fx_to_onnxscript`, and assuming the two graphs are in sync.

I'm fine if we are not there yet. It might be a little back and forth, but for the current state I feel it might be better to move these code inside `export_fx_to_onnxscript`. This pass is only needed when `op_level_debug=True`. Bringing it local makes the "in sync assumption" part more natural.

Reply:
So I merged the Transformer into ShapeInference pass. And embed the propagate `node_with_static_shape` into original node.meta["node_with_static_shape"]. We can work on the naming of this metadata, but I think this is the best solution we can have now. WDYT

Reply:
The downside is that we still have that sync assumption of two graphs. I tried to FakeTensorProp value into node.meta["real_shape"] instead of node.meta["val"], and it doesn't work. I guess if we really want same module in same module out on this pass, we might have to write an Interpreter.

Reply:
> tried to FakeTensorProp value into node.meta["real_shape"] instead of node.meta["val"], and it doesn't work.

Feel free to call